docs/_writing-tests/general-guidelines.md

```diff
@@ -98,6 +98,30 @@ This is not, however, to discourage testing of edge cases or
 interactions between multiple features; such tests are an essential
 part of ensuring interoperability of the web platform.
 
+Tests should pass when the feature under test exposes the expected behavior,
+and they should fail when the feature under test is not implemented or is
+implemented incorrectly. Tests should not rely on unrelated features if doing
+so causes failures in the latest stable release of [Apple
+Safari][apple-safari], [Google Chrome][google-chrome], [Microsoft
+Edge][micosoft-edge], or [Mozilla Firefox][mozilla-firefox]. They should,
+therefore, not rely on any features aside from the one under test unless they
+are supported in all four browsers.
```

Comment:
May be worth encouraging that when a contributor runs into an unrelated feature not working as expected in the 4 browsers, that they write a separate test for that, or at least report an issue about it so somebody else can follow up.